Could not pass event BlockBreakEvent to Slimefun v4.1.9

## Description
Describe your Issue AS DETAILED AS POSSIBLE.
Could not pass event BlockBreakEvent to Slimefun v4.1.9

## Server Log
Upload your FULL SERVER LOG (/logs/latest.log) to http://pastebin.com/ and paste your link in here.
Without this File, we may not be able to help you at all.
Send if need

## Evidence / Proof
Attach as many Screenshots or Videos if possible.
Anything that helps us understand the Issue better, is gonna get your Issue fixed even faster.
```
[11:30:11] [Server thread/ERROR]: Could not pass event BlockBreakEvent to Slimefun v4.1.9
org.bukkit.event.EventException: null
	at org.bukkit.plugin.EventExecutor$1.execute(EventExecutor.java:46) ~[patched_1.12.jar:git-Paper-1147]
	at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:74) ~[patched_1.12.jar:git-Paper-1147]
	at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:62) ~[patched_1.12.jar:git-Paper-1147]
	at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:514) ~[patched_1.12.jar:git-Paper-1147]
	at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:499) ~[patched_1.12.jar:git-Paper-1147]
	at net.minecraft.server.v1_12_R1.PlayerInteractManager.breakBlock(PlayerInteractManager.java:289) ~[patched_1.12.jar:git-Paper-1147]
	at net.minecraft.server.v1_12_R1.PlayerInteractManager.a(PlayerInteractManager.java:191) ~[patched_1.12.jar:git-Paper-1147]
	at net.minecraft.server.v1_12_R1.PlayerConnection.a(PlayerConnection.java:871) ~[patched_1.12.jar:git-Paper-1147]
	at net.minecraft.server.v1_12_R1.PacketPlayInBlockDig.a(SourceFile:40) ~[patched_1.12.jar:git-Paper-1147]
	at net.minecraft.server.v1_12_R1.PacketPlayInBlockDig.a(SourceFile:10) ~[patched_1.12.jar:git-Paper-1147]
	at net.minecraft.server.v1_12_R1.PlayerConnectionUtils.lambda$ensureMainThread$0(PlayerConnectionUtils.java:14) ~[patched_1.12.jar:git-Paper-1147]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_131]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_131]
	at net.minecraft.server.v1_12_R1.SystemUtils.a(SourceFile:46) [patched_1.12.jar:git-Paper-1147]
	at net.minecraft.server.v1_12_R1.MinecraftServer.D(MinecraftServer.java:842) [patched_1.12.jar:git-Paper-1147]
	at net.minecraft.server.v1_12_R1.DedicatedServer.D(DedicatedServer.java:423) [patched_1.12.jar:git-Paper-1147]
	at net.minecraft.server.v1_12_R1.MinecraftServer.C(MinecraftServer.java:766) [patched_1.12.jar:git-Paper-1147]
	at net.minecraft.server.v1_12_R1.MinecraftServer.run(MinecraftServer.java:664) [patched_1.12.jar:git-Paper-1147]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_131]
Caused by: java.lang.IndexOutOfBoundsException: Index: 4, Size: 4
	at java.util.ArrayList.rangeCheck(ArrayList.java:653) ~[?:1.8.0_131]
	at java.util.ArrayList.get(ArrayList.java:429) ~[?:1.8.0_131]
	at me.mrCookieSlime.Slimefun.Setup.SlimefunSetup$19.onBlockBreak(SlimefunSetup.java:1429) ~[?:?]
	at me.mrCookieSlime.Slimefun.listeners.ToolListener.onBlockBreak(ToolListener.java:223) ~[?:?]
	at com.destroystokyo.paper.event.executor.asm.generated.GeneratedEventExecutor641.execute(Unknown Source) ~[?:?]
	at org.bukkit.plugin.EventExecutor$1.execute(EventExecutor.java:44) ~[patched_1.12.jar:git-Paper-1147]
	... 18 more

```
## Environment
We need the EXACT VERSIONS of the Software you are using.
This includes your Minecraft Version, your CS-CoreLib version and your Slimefun Version.
If you cannot figure out what versions you are running, then please run /sf versions and
show us the output of that command.
```
Paper v1_12_R1
CS-CoreLib v1.5.16
Slimefun v4.1.9
Installed Addons (4)
ExtraGear v1.1.1
SlimefunLuckyBlocks v1.1
ExoticGarden v1.6.3
ElectricSpawners v1.0
```

--- a/src/me/mrCookieSlime/Slimefun/Objects/SlimefunItem/EnderTalisman.java
+++ b/src/me/mrCookieSlime/Slimefun/Objects/SlimefunItem/EnderTalisman.java
@@ -9,11 +9,11 @@ import org.bukkit.potion.PotionEffect;
 
 public class EnderTalisman extends SlimefunItem {
 	
-	boolean consumed;
-	boolean cancel;
-	PotionEffect[] effects;
-	String suffix;
-	int chance;
+	private boolean consumed;
+	private boolean cancel;
+	private PotionEffect[] effects;
+	private String suffix;
+	private int chance;
 
 	public EnderTalisman(Talisman parent) {
 		super(Categories.TALISMANS_2, parent.upgrade(), "ENDER_" + parent.getID(), RecipeType.MAGIC_WORKBENCH, new ItemStack[] {SlimefunItem.getItem("ENDER_LUMP_3"), null, SlimefunItem.getItem("ENDER_LUMP_3"), null, parent.getItem(), null, SlimefunItem.getItem("ENDER_LUMP_3"), null, SlimefunItem.getItem("ENDER_LUMP_3")}, parent.upgrade());

--- a/src/me/mrCookieSlime/Slimefun/Objects/SlimefunItem/EnderTalisman.java
+++ b/src/me/mrCookieSlime/Slimefun/Objects/SlimefunItem/EnderTalisman.java
@@ -7,27 +7,30 @@ import me.mrCookieSlime.Slimefun.api.Slimefun;
 import org.bukkit.inventory.ItemStack;
 import org.bukkit.potion.PotionEffect;
 
+/**
+ * @since 4.0
+ */
 public class EnderTalisman extends SlimefunItem {
-	
-	private boolean consumed;
+
+	private String suffix;
+	private boolean consumable;
 	private boolean cancel;
 	private PotionEffect[] effects;
-	private String suffix;
 	private int chance;
 
 	public EnderTalisman(Talisman parent) {
 		super(Categories.TALISMANS_2, parent.upgrade(), "ENDER_" + parent.getID(), RecipeType.MAGIC_WORKBENCH, new ItemStack[] {SlimefunItem.getItem("ENDER_LUMP_3"), null, SlimefunItem.getItem("ENDER_LUMP_3"), null, parent.getItem(), null, SlimefunItem.getItem("ENDER_LUMP_3"), null, SlimefunItem.getItem("ENDER_LUMP_3")}, parent.upgrade());
-		this.consumed = parent.isConsumable();
+		this.consumable = parent.isConsumable();
 		this.cancel = parent.isEventCancelled();
 		this.suffix = parent.getSuffix();
 		this.effects = parent.getEffects();
 		this.chance = parent.getChance();
 		Slimefun.addHint("ENDER_" + parent.getID(), "&eEnder Talismans have the advantage", "&eof still working while they", "&eare in your Ender Chest");
 	}
-	
-	public PotionEffect[] getEffects()	{		return this.effects;	}
-	public boolean isConsumable()	 	{		return this.consumed;	}
-	public boolean isEventCancelled() 	{		return this.cancel;		}
+
 	public String getSuffix() 			{		return this.suffix;		}
+	public boolean isConsumable()	 	{		return this.consumable;	}
+	public boolean isEventCancelled() 	{		return this.cancel;		}
+	public PotionEffect[] getEffects()	{		return this.effects;	}
 	public int getChance()				{		return this.chance;		}
 }

--- a/src/me/mrCookieSlime/Slimefun/Objects/SlimefunItem/Talisman.java
+++ b/src/me/mrCookieSlime/Slimefun/Objects/SlimefunItem/Talisman.java
@@ -25,11 +25,11 @@ import org.bukkit.potion.PotionEffect;
 
 public class Talisman extends SlimefunItem {
 	
-	boolean consumed;
-	boolean cancel;
-	PotionEffect[] effects;
-	String suffix;
-	int chance;
+	private boolean consumed;
+	private boolean cancel;
+	private PotionEffect[] effects;
+	private String suffix;
+	private int chance;
 
 	public Talisman(ItemStack item, String id, ItemStack[] recipe, boolean consumable, boolean cancelEvent, String messageSuffix, PotionEffect... effects) {
 		super(Categories.TALISMANS_1, item, id, RecipeType.MAGIC_WORKBENCH, recipe, new CustomItem(item, consumable ? 4: 1));

--- a/src/me/mrCookieSlime/Slimefun/Objects/SlimefunItem/Talisman.java
+++ b/src/me/mrCookieSlime/Slimefun/Objects/SlimefunItem/Talisman.java
@@ -23,26 +23,28 @@ import org.bukkit.event.player.PlayerEvent;
 import org.bukkit.inventory.ItemStack;
 import org.bukkit.potion.PotionEffect;
 
+/**
+ * @since 4.0
+ */
 public class Talisman extends SlimefunItem {
-	
-	private boolean consumed;
-	private boolean cancel;
-	private PotionEffect[] effects;
+
 	private String suffix;
-	private int chance;
+	private boolean consumable = true;
+	private boolean cancel = true;
+	private PotionEffect[] effects = new PotionEffect[0];
+	private int chance = 100;
 
 	public Talisman(ItemStack item, String id, ItemStack[] recipe, boolean consumable, boolean cancelEvent, String messageSuffix, PotionEffect... effects) {
 		super(Categories.TALISMANS_1, item, id, RecipeType.MAGIC_WORKBENCH, recipe, new CustomItem(item, consumable ? 4: 1));
-		this.consumed = consumable;
+		this.consumable = consumable;
 		this.cancel = cancelEvent;
 		this.suffix = messageSuffix;
 		this.effects = effects;
-		this.chance = 100;
 	}
 	
 	public Talisman(ItemStack item, String id, ItemStack[] recipe, boolean consumable, boolean cancelEvent, String messageSuffix, int chance, PotionEffect... effects) {
 		super(Categories.TALISMANS_1, item, id, RecipeType.MAGIC_WORKBENCH, recipe, new CustomItem(item, consumable ? 4: 1));
-		this.consumed = consumable;
+		this.consumable = consumable;
 		this.cancel = cancelEvent;
 		this.suffix = messageSuffix;
 		this.effects = effects;
@@ -51,23 +53,21 @@ public class Talisman extends SlimefunItem {
 	
 	public Talisman(ItemStack item, String id, ItemStack[] recipe, String messageSuffix, int chance, PotionEffect... effects) {
 		super(Categories.TALISMANS_1, item, id, RecipeType.MAGIC_WORKBENCH, recipe, item);
-		this.consumed = true;
-		this.cancel = true;
 		this.suffix = messageSuffix;
 		this.effects = effects;
 		this.chance = chance;
 	}
-	
-	public PotionEffect[] getEffects()	{		return this.effects;	}
-	public boolean isConsumable()	 	{		return this.consumed;	}
-	public boolean isEventCancelled() 	{		return this.cancel;		}
+
 	public String getSuffix() 			{		return this.suffix;		}
+	public boolean isConsumable()	 	{		return this.consumable;	}
+	public boolean isEventCancelled() 	{		return this.cancel;		}
+	public PotionEffect[] getEffects()	{		return this.effects;	}
 	public int getChance()				{		return this.chance;		}
 
 	public static boolean checkFor(Event e, SlimefunItem talisman) {
 		if (talisman != null) {
 			if (talisman instanceof Talisman) {
-				boolean message = ((Talisman) talisman).getSuffix().equalsIgnoreCase("") ? false: true;
+				boolean message = !((Talisman) talisman).getSuffix().equalsIgnoreCase("");
 				if (SlimefunStartup.chance(100, ((Talisman) talisman).getChance())) {
 					Player p = null;
 					

--- a/src/me/mrCookieSlime/Slimefun/Objects/SlimefunItem/Talisman.java
+++ b/src/me/mrCookieSlime/Slimefun/Objects/SlimefunItem/Talisman.java
@@ -80,7 +80,7 @@ public class Talisman extends SlimefunItem {
 					boolean pass = true;
 					
 					for (PotionEffect effect: ((Talisman) talisman).getEffects()) {
-						if (effect != null) if (p.hasPotionEffect(effect.getType())) pass = false;
+						if (effect != null && p.hasPotionEffect(effect.getType())) pass = false;
 					}
 					
 					if (pass) {

--- a/src/me/mrCookieSlime/Slimefun/Setup/SlimefunSetup.java
+++ b/src/me/mrCookieSlime/Slimefun/Setup/SlimefunSetup.java
@@ -1459,10 +1459,11 @@ public class SlimefunSetup {
 					if (e.getBlock().getType().equals(Material.SKULL)) return true;
 
 					int j = -1;
-					for (int i = 0; i < e.getBlock().getDrops().size(); i++) {
-						if (((List<ItemStack>) e.getBlock().getDrops()).get(i) != null) {
+					List<ItemStack> dropsList = (List<ItemStack>) e.getBlock().getDrops();
+					for (int i = 0; i < dropsList.size(); i++) {
+						if (dropsList.get(i) != null) {
 							j++;
-							drops.add(e.getBlock().getType().toString().endsWith("_ORE") ? new CustomItem(((List<ItemStack>) e.getBlock().getDrops()).get(i), fortune): ((List<ItemStack>) e.getBlock().getDrops()).get(i));
+							drops.add(e.getBlock().getType().toString().endsWith("_ORE") ? new CustomItem(dropsList.get(i), fortune): dropsList.get(i));
 							if (RecipeCalculator.getSmeltedOutput(drops.get(i).getType()) != null) {
 								e.getBlock().getWorld().playEffect(e.getBlock().getLocation(), Effect.MOBSPAWNER_FLAMES, 1);
 								drops.set(j, new CustomItem(RecipeCalculator.getSmeltedOutput(drops.get(i).getType()), drops.get(i).getAmount()));

--- a/src/me/mrCookieSlime/Slimefun/Setup/SlimefunSetup.java
+++ b/src/me/mrCookieSlime/Slimefun/Setup/SlimefunSetup.java
@@ -1434,7 +1434,6 @@ public class SlimefunSetup {
 
 		new SlimefunItem(Categories.TECH_MISC, SlimefunItems.SOLAR_PANEL, "SOLAR_PANEL", RecipeType.ENHANCED_CRAFTING_TABLE,
 		new ItemStack[] {new ItemStack(Material.GLASS), new ItemStack(Material.GLASS), new ItemStack(Material.GLASS), SlimefunItems.SILICON, SlimefunItems.SILICON, SlimefunItems.SILICON, SlimefunItems.FERROSILICON, SlimefunItems.FERROSILICON, SlimefunItems.FERROSILICON})
-
 		.register(true);
 
 		new SolarHelmet(Categories.TECH, SlimefunItems.SOLAR_HELMET, "SOLAR_HELMET", RecipeType.ENHANCED_CRAFTING_TABLE,

--- a/src/me/mrCookieSlime/Slimefun/listeners/ItemPickupListener.java
+++ b/src/me/mrCookieSlime/Slimefun/listeners/ItemPickupListener.java
@@ -9,7 +9,7 @@ import org.bukkit.event.player.PlayerPickupItemEvent;
 /**
  * Listens to the ItemPickup events to prevent it if the item has the "no_pickup" metadata or is an ALTAR_PROBE.
  *
- * @since 4.1.12
+ * @since 4.1.11
  */
 public class ItemPickupListener implements Listener {
 

--- a/src/me/mrCookieSlime/Slimefun/listeners/ItemPickupListener_1_12.java
+++ b/src/me/mrCookieSlime/Slimefun/listeners/ItemPickupListener_1_12.java
@@ -11,7 +11,7 @@ import org.bukkit.event.inventory.InventoryPickupItemEvent;
  *
  * This Listener uses the new {@link EntityPickupItemEvent} due to the deprecation of {@link org.bukkit.event.player.PlayerPickupItemEvent}.
  *
- * @since 4.1.12
+ * @since 4.1.11
  */
 public class ItemPickupListener_1_12 implements Listener {
 

--- a/src/me/mrCookieSlime/Slimefun/listeners/TalismanListener.java
+++ b/src/me/mrCookieSlime/Slimefun/listeners/TalismanListener.java
@@ -4,12 +4,14 @@ import java.util.ArrayList;
 import java.util.List;
 import java.util.Set;
 
+import me.mrCookieSlime.CSCoreLibPlugin.general.Inventory.Item.CustomItem;
 import me.mrCookieSlime.Slimefun.SlimefunStartup;
 import me.mrCookieSlime.Slimefun.Objects.SlimefunItem.SlimefunItem;
 import me.mrCookieSlime.Slimefun.Objects.SlimefunItem.Talisman;
 import me.mrCookieSlime.Slimefun.Setup.SlimefunManager;
 import me.mrCookieSlime.Slimefun.api.Slimefun;
 
+import org.bukkit.Material;
 import org.bukkit.Sound;
 import org.bukkit.enchantments.Enchantment;
 import org.bukkit.entity.LivingEntity;
@@ -18,12 +20,14 @@ import org.bukkit.entity.Projectile;
 import org.bukkit.event.EventHandler;
 import org.bukkit.event.EventPriority;
 import org.bukkit.event.Listener;
+import org.bukkit.event.block.BlockBreakEvent;
 import org.bukkit.event.enchantment.EnchantItemEvent;
 import org.bukkit.event.entity.EntityDamageByEntityEvent;
 import org.bukkit.event.entity.EntityDamageEvent;
 import org.bukkit.event.entity.EntityDamageEvent.DamageCause;
 import org.bukkit.event.player.PlayerItemBreakEvent;
 import org.bukkit.event.player.PlayerToggleSprintEvent;
+import org.bukkit.inventory.ItemStack;
 import org.bukkit.potion.PotionEffect;
 import org.bukkit.potion.PotionEffectType;
 import org.bukkit.util.Vector;
@@ -100,6 +104,35 @@ public class TalismanListener implements Listener {
 				}
 				
 				e.getItem().addUnsafeEnchantment(Enchantment.LOOT_BONUS_BLOCKS, SlimefunStartup.randomize(3) + 3);
+			}
+		}
+	}
+
+	/**
+	 *
+	 * @param e BlockBreakEvent
+	 * @since 4.2.0
+	 */
+	@EventHandler
+	public void onBlockBreak(BlockBreakEvent e) {
+		List<ItemStack> drops = new ArrayList<ItemStack>();
+		ItemStack item = e.getPlayer().getInventory().getItemInMainHand();
+		int fortune = 1;
+
+		if (item != null) {
+			if (item.getEnchantments().containsKey(Enchantment.LOOT_BONUS_BLOCKS) && !item.getEnchantments().containsKey(Enchantment.SILK_TOUCH)) {
+				fortune = SlimefunStartup.randomize(item.getEnchantmentLevel(Enchantment.LOOT_BONUS_BLOCKS) + 2) - 1;
+				if (fortune <= 0) fortune = 1;
+				fortune = (e.getBlock().getType() == Material.LAPIS_ORE ? 4 + SlimefunStartup.randomize(5) : 1) * (fortune + 1);
+			}
+
+			if (!item.getEnchantments().containsKey(Enchantment.SILK_TOUCH) && e.getBlock().getType().toString().endsWith("_ORE")) {
+				if (Talisman.checkFor(e, SlimefunItem.getByID("MINER_TALISMAN"))) {
+					if (drops.isEmpty()) drops = (List<ItemStack>) e.getBlock().getDrops();
+					for (ItemStack drop : new ArrayList<ItemStack>(drops)) {
+						if (!drop.getType().isBlock()) drops.add(new CustomItem(drop, fortune * 2));
+					}
+				}
 			}
 		}
 	}

--- a/src/me/mrCookieSlime/Slimefun/listeners/ToolListener.java
+++ b/src/me/mrCookieSlime/Slimefun/listeners/ToolListener.java
@@ -216,15 +216,6 @@ public class ToolListener implements Listener {
 			}
 		}
 		
-		if (!item.getEnchantments().containsKey(Enchantment.SILK_TOUCH) && e.getBlock().getType().toString().endsWith("_ORE")) {
-			if (Talisman.checkFor(e, SlimefunItem.getByName("MINER_TALISMAN"))) {
-				if (drops.isEmpty()) drops = (List<ItemStack>) e.getBlock().getDrops();
-				for (ItemStack drop: new ArrayList<ItemStack>(drops)) {
-					if (!drop.getType().isBlock()) drops.add(new CustomItem(drop, fortune * 2));
-				}
-			}
-		}
-		
 		if (!drops.isEmpty()) {
 			e.getBlock().setType(Material.AIR);
 			for (ItemStack drop: drops) {

